crates/mysten-service/src/server_timing.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use axum::extract::Request;
+use axum::middleware::Next;
+use axum::response::Response;
+use mysten_metrics::{add_server_timing, get_server_timing, with_new_server_timing};
+use simple_server_timing_header::Timer;
+
+pub async fn server_timing_middleware(request: Request, next: Next) -> Response {
+    with_new_server_timing(async move {
+        let mut response = next.run(request).await;
+        add_server_timing("finish_request");
```

Comment:
Do you want to have a "start" time before the request starts?

Reply:
it would effectively always be 0, since we can only start counting from the point at which we enter this middleware, so it probably wouldn't be useful

Reply:
I wasn't sure if we wanted to make sure it was initialized before calling into the handler or not

Reply:
I see - it's eagerly initialized in `with_new_server_timing` so there is no need to worry about that. This final `add_server_timing` call just ensures that, if nothing else, the total duration of the handler is always measured